tower-layer/src/lib.rs

```diff
@@ -11,7 +11,7 @@
 //! A layer decorates an service and provides additional functionality. It
 //! allows other services to be composed with the service that implements layer.
 //!
-//! A middleware implements the [`Layer`] and [`Service`] trait.
+//! A middleware implements the [`Layer`] and [`Service`](tower_service::Service) trait.
```

Comment:
This still won't work until https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md is merged.